app/controllers/posts_controller.rb

```diff
@@ -1,15 +1,30 @@
 class PostsController < ApplicationController
-  before_action :set_user
+  before_action :set_user, only: %i[index show new create]
   before_action :set_post, only: [:show]
 
   def index
     @posts = @user.posts.paginate(page: params[:page], per_page: 2)
   end
 
   def show
-    @posts = @user.posts.find(params[:id])
-    @comments = @posts.comments
-    @comments_user = @comments.map(&:user)
+    @like = Like.new
+  end
+
+  def new
+    @post = @user.posts.new
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to user_posts_path(@user)
+    else
+      flash[:error] = 'Error: Post could not be saved'
+      flash[:error_details] = @post.errors.full_messages.join(', ')
+      render :new
+    end
   end
```

Comment:
- [ ] @ometman Great work adding a method to add new posts, however, we are supposed to use the `current_user` method created on the application controller to create a new post on behalf of the current user, kindly consider using the method to get your user.
![image](https://github.com/ometman/blogtalk/assets/61408860/feb94455-32a6-4a05-bcb9-417e080ba8c5)